src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1709,38 +1709,39 @@ private bool SkipComment()
 
         private bool SkipSingleLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
         {
-            idx = localBuffer.IndexOfAny(JsonConstants.LineFeed, JsonConstants.CarriageReturn);
-            if (idx == -1)
+            idx = localBuffer.IndexOfAny(JsonConstants.CarriageReturn, JsonConstants.LineFeed);
+            if (idx != -1)
             {
-                if (IsLastSpan)
+                if (localBuffer[idx] == JsonConstants.LineFeed)
                 {
-                    idx = localBuffer.Length;
-                    // Assume everything on this line is a comment and there is no more data.
-                    _bytePositionInLine += 2 + localBuffer.Length;
-                    goto Done;
+                    goto EndOfComment;
                 }
-                return false;
-            }
-
-            // If we have encountered \r, check and advance index if next character is \n
-            if (localBuffer[idx] == JsonConstants.CarriageReturn)
-            {
-                if (idx < localBuffer.Length - 1)
+                if ((idx + 1) < localBuffer.Length)
```

Comment:
Could this potentially integer overflow? Maybe re-write it as `if (idx < localBuffer.Length - 1)`